### DISCUSSION TITLE
Lots of Error [Semantical Error] The annotation with phpstorm

### DISCUSSION THREAD

**Discussion by wnunezc** [symfony/symfony#44454] [Q&A]
Hello guys i have excecute reformat code in phpstorm; after this i am getting a number of errors about `[Semantical Error] The annotation`   Is there any command from composer or symfony to rebuild this?

![image](https://user-images.githubusercontent.com/1207306/144650826-19ef3c6d-3449-4cee-be0e-2104c67ecbf3.png)

**Comment by stof** [comment]
This is not an issue in Symfony:

- the annotation parser is maintained at https://github.com/doctrine/annotations/
- the ORM (from which the annotation comes from is maintained at https://github.com/doctrine/orm
- the PHPStorm issue tracker is at https://youtrack.jetbrains.com/issues/WI (if this only happens when running PHPStorm's formatting logic, it is likely a bug in it).

There is noting from Symfony in this report AFAICT.

Thus, you don't provide any info to reproduce it.

**Reply by wnunezc** [reply]
> This is not an issue in Symfony:
> 
> * the annotation parser is maintained at https://github.com/doctrine/annotations/
> * the ORM (from which the annotation comes from is maintained at https://github.com/doctrine/orm
> * the PHPStorm issue tracker is at https://youtrack.jetbrains.com/issues/WI (if this only happens when running PHPStorm's formatting logic, it is likely a bug in it).
> 
> There is noting from Symfony in this report AFAICT.
> 
> Thus, you don't provide any info to reproduce it.

to reproduce only need to click in  reformat code and excecute it ... it will perform a clean code automatically that i think broke the semantic of the bundles ... and in general kernel symfony

**Reply by stof** [reply]
> to reproduce only need to click in reformat code and excecute it

Reformatting which code ? You haven't provided it.
and if this happens when running a **PHPStorm** tooling, this is a bug in PHPStorm, not in Symfony.

And without seeing what happens to the code, we cannot even help you identify what got broken.

**Reply by wnunezc** [reply]
It seems to me that you are not reading correctly; the option to reformat the code is done on the project ... and he optimizes, reformate and cleans the whole project ... there is no code other than the installation of a project with any name ... you don't need anything else to reproduce this ... I only installed a project as indicated in the documentation ...

**Reply by stof** [reply]
> I only installed a project as indicated in the documentation

As indicated in **which** documentation ? And even if you talk about the official Symfony documentation, which page are you following ? The official documentation is big.

Your error message indicates an error in `App\Entity\User`, which is code from the application, not from Symfony itself. And as I don't know which application you are talking about, I cannot reproduce it.

**Reply by wnunezc** [reply]
ok, I'm not going to waste any more time if you don't know what I'm talking about ... you just need to tell me that you don't know how to install synfony ... it seems like a joke ...

**Comment by stof** [comment]
One guess is that PHPStorm's reformatting might remove unused use statements, which would remove the use statements relied on by annotations (as those are used inside the doc comments). the solution for that is to install the [PHP Annotations plugin](https://plugins.jetbrains.com/plugin/7320-php-annotations) which adds support for annotations in PHPStorm (and so will detect that the use statements are not unused)

**Reply by wnunezc** [reply]
ok I will do it for the next one I have deleted the project and I will have to start from 0 .. I am following a tutorial of version 5.0.2 that I found on youtube in Spanish and although there are some different things it is quite similar ...
